price_feed_poller.py

```diff
@@ -9,37 +9,44 @@
 from config import load_config
 
 # Post a data request to the post_dr method of the WRB contract
-def handle_requestUpdate(w3, pricefeedcontract, wrbcontract, account_addr, gas, gas_price):
+def handle_requestUpdate(
+    w3,
+    pricefeedcontract,
+    wrbcontract,
+    account_addr,
+    gas,
+    gas_price,
+    tx_waiting_timeout_secs,
+    tx_polling_latency_secs
+  ):
 
-    # Check that the accout has enough balance
-    balance = w3.eth.getBalance(account_addr)
-    if balance == 0:
-        raise Exception("Account does not have any funds")
+    try:
+      # Check that the accout has enough balance
+      balance = w3.eth.getBalance(account_addr)
 
-    print(f"Got {balance} wei")
-    print(f"Gas price: {gas_price}")
-    reward = wrbcontract.functions.estimateGasCost(gas_price).call()
+      if balance == 0:
+          raise Exception("Account does not have any funds")
+
+      print(f"Got {balance} wei")
+      print(f"Gas price: {gas_price}")
+
+      reward = wrbcontract.functions.estimateGasCost(gas_price).call()
 
-    # Hardcoded gas since it does not estimate well
-    try:
       dr_id = pricefeedcontract.functions.requestUpdate().transact({
         "from": account_addr,
         "gas": gas,
         "gasPrice": gas_price,
         "value": reward
       })
-    except:
-      print(f"Failed when calling to", account_addr, ".requestUpdate(). Retrying in next iteration.")
-      return False
 
-    try:     
       # Get receipt of the transaction   
-      receipt = w3.eth.waitForTransactionReceipt(dr_id, 120, 5)
+      try:
+        receipt = w3.eth.waitForTransactionReceipt(dr_id, tx_waiting_timeout_secs, tx_polling_latency_secs)
+      except exceptions.TimeExhausted:
+        print(f"Transaction for requesting update on {pricefeedcontract.address} is taking too long. Retrying in next iteration.")
 
-    except exceptions.TimeExhausted:
-      print(
-        f"Transaction timeout reached and dr post not included in the block. Retrying in next iteration."
-      )
+    except:
+      print(f"Failed when trying to update request on {pricefeedcontract.address}. Retrying in next iteration.")
```

Comment:
This terminal exception handler is silently hiding the reason for the exception. Let's take this chance to print the exception for easier troubleshooting:
```suggestion
    except Exception as ex:
      print(f"Failed when trying to update request on {pricefeedcontract.address}. Retrying in next iteration. Exception was: {ex}")
```